pac4j-ldap/src/test/java/org/pac4j/ldap/credentials/authenticator/LdapAuthenticatorTests.java

```diff
@@ -0,0 +1,133 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.ldap.credentials.authenticator;
+
+import org.junit.*;
+import org.ldaptive.auth.Authenticator;
+import org.pac4j.core.exception.BadCredentialsException;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+import org.pac4j.ldap.profile.LdapProfile;
+import org.pac4j.ldap.test.tools.AuthenticatorGenerator;
+import org.pac4j.ldap.test.tools.LdapServer;
+
+import java.util.Collection;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests the {@link LdapAuthenticator}.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.0
+ */
+public class LdapAuthenticatorTests {
+
+    private final static String CLIENT_NAME = "clientName";
+    private final static String BAD_USERNAME = "michael";
```

Comment:
I was looking for a bad username example and I was thinking of you: so here you are in the source code...